### PR TITLE
Remove `psr/log` dependency, upgrade psalm configuration to latest format

### PR DESCRIPTION
**Pull request by adamturcsan**
Ignore and remove composer.lock from git as it prevents installing this lib into various environments.

Also remove psr/log and its version constraint from composer.json as it was too specific and quite frankly unnecessary as this lib doesn't even use it.

Fix psalm issue of PHPUnit mock not following annotated union type interface. Added new common interface that extends two other ones. It may be unnecessarily in the main project namespace.

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -1,5790 +0,0 @@
-{
```
Comment (Ocramius) [inline composer.lock]: `composer.lock` should be preserved: it doesn't affect downstream, but it does affect the stability of our CI environment.

File: src/Mezzio/Storageless/SessionThatIsIdentifierAware.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PSR7Sessions\Mezzio\Storageless;
+
+use Mezzio\Session\SessionIdentifierAwareInterface;
+use Mezzio\Session\SessionInterface;
+
+interface SessionThatIsIdentifierAware extends SessionInterface, SessionIdentifierAwareInterface {}
```
Comment (Ocramius) [inline src/Mezzio/Storageless/SessionThatIsIdentifierAware.php]: IMO worth putting this in the test suite only

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 vendor
 composer.phar
+composer.lock
```
Comment (Ocramius) [inline .gitignore]: This line should be removed: as mentioned before, `composer.lock` is to be retained.

File: composer.lock (hunk 1 above)
Comment (adamturcsan) [reply composer.lock]: Sorry for misinterpreting the situation. It was enough to remove psr/log from composer.json, composer.lock didn't do anything of preventing anything :) I've added it back.

File: src/Mezzio/Storageless/SessionThatIsIdentifierAware.php (hunk 2 above)
Comment (adamturcsan) [reply src/Mezzio/Storageless/SessionThatIsIdentifierAware.php]: Moved it to `PSR7SessionsTest\Mezzio\Storageless;`

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @adamturcsan!

**Pull request merged**